packages/esbuild/src/executors/esbuild/lib/build-esbuild-options.ts

```diff
@@ -123,6 +127,50 @@ export function buildEsbuildOptions(
   return esbuildOptions;
 }
 
+/**
+ * When using TS project references we need to map the paths to the referenced projects.
+ * This is necessary because esbuild does not support project references out of the box.
+ * @param context ExecutorContext
+ */
+export function createPathsFromTsConfigReferences(
+  context: ExecutorContext
+): Record<string, string[]> {
+  const {
+    findAllProjectNodeDependencies,
+  } = require('nx/src/utils/project-graph-utils');
+
+  const deps = findAllProjectNodeDependencies(context.projectName);
+  const tsConfig = readJsonFile(
+    joinPathFragments(context.root, 'tsconfig.json')
+  );
+  const referencesAsPaths = new Set(
+    tsConfig.references.map((ref) => joinPathFragments(workspaceRoot, ref.path))
```

Comment:
`ref.path` could be a path pointing to a tsconfig file. We're using this `Set` as if it's always a directory, but even though that's what we generate, it could be modified by the user and point to a file. We should normalize it to make sure it's a directory.